oak-core/src/main/java/org/apache/jackrabbit/oak/query/SQL2Parser.java

```diff
@@ -290,6 +294,14 @@ private SelectorImpl parseSelector() throws ParseException {
 
         return factory.selector(nodeTypeInfo, selectorName);
     }
+
+    private long readBase10WholeNumber() throws ParseException {
+        String label = readName();
+        if (!label.matches("\\d*") || label.isEmpty() || label.length() > 19 /* Length of MAX_LONG */) {
```

Comment:
I don't think the "if" condition is needed; instead I would use catch (NumberFormatException nfe) (same as below).